pkg/cmd/server/start/start_node.go

```diff
@@ -225,17 +226,17 @@ func (o NodeOptions) CreateNodeConfig() error {
 	return nil
 }
 
-func RunSDNController(config configapi.NodeConfig) {
+func RunSDNController(config configapi.NodeConfig, nodeConfig *kubernetes.NodeConfig) {
 	if config.NetworkPluginName == osdn.NetworkPluginName() {
-		kclient, _, err := configapi.GetKubeClient(config.MasterKubeConfig)
-		if err != nil {
-			glog.Fatal("Failed to get kube client for SDN")
-		}
 		oclient, _, err := configapi.GetOpenShiftClient(config.MasterKubeConfig)
 		if err != nil {
 			glog.Fatal("Failed to get kube client for SDN")
 		}
-		osdn.Node(*oclient, *kclient, config.NodeName, "")
+		ch := make(chan struct{})
+		nodeConfig.KubeletConfig.StartUpdates = ch
+		go osdn.Node(oclient, nodeConfig.Client, config.NodeName, "", ch)
+		// no plugin necessary
+		nodeConfig.KubeletConfig.NetworkPluginName = ""
```

Comment:
This is ready for a second try - it preserves the plugin name, moves the systemd ready notification to the options.Start* methods, and ensures the kubelet is run in a goroutine so as to not block start.

----- Original Message -----
> >  		oclient, _, err := configapi.GetOpenShiftClient(config.MasterKubeConfig)
> >  		if err != nil {
> >  			glog.Fatal("Failed to get kube client for SDN")
> >  		}
> > -		osdn.Node(*oclient, *kclient, config.NodeName, "")
> > +		ch := make(chan struct{})
> > +		nodeConfig.KubeletConfig.StartUpdates = ch
> > +		go osdn.Node(oclient, nodeConfig.Client, config.NodeName, "", ch)
> > +		// no plugin necessary
> > +		nodeConfig.KubeletConfig.NetworkPluginName = ""
> 
> nm, that line was fixing a bug where the arg wasn't persisted when
> --write-config was used
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/2984/files#r32060332